libkbfs/keybase_service_base.go

```diff
@@ -440,16 +448,58 @@ func (k *KeybaseServiceBase) LoadUserPlusKeys(ctx context.Context,
 // LoadTeamPlusKeys implements the KeybaseService interface for
 // KeybaseServiceBase.
 func (k *KeybaseServiceBase) LoadTeamPlusKeys(
-	ctx context.Context, tid keybase1.TeamID) (TeamInfo, error) {
+	ctx context.Context, tid keybase1.TeamID, desiredKeyGen KeyGen,
+	desiredUser keybase1.UserVersion, desiredRole keybase1.TeamRole) (
+	TeamInfo, error) {
+	if desiredRole != keybase1.TeamRole_NONE &&
+		desiredRole != keybase1.TeamRole_WRITER &&
+		desiredRole != keybase1.TeamRole_READER {
+		panic("KBFS shouldn't care about non-writer/reader roles")
+	}
+
 	cachedTeamInfo := k.getCachedTeamInfo(tid)
 	if cachedTeamInfo.Name != libkb.NormalizedUsername("") {
-		return cachedTeamInfo, nil
+		// If the cached team info doesn't satisfy our desires, don't
+		// use it.
+		satisfiesDesires := true
+		if desiredKeyGen >= FirstValidKeyGen {
```

Comment:
This might be an obvious comment, but it saves having to imagine the scenario represented by the code.

`// If the desiredKeyGen is beyond the cached latest team info, then our cached key must be out of date.`